C/plugins/storage/postgres/connection.cpp

```diff
@@ -3312,3 +3312,240 @@ const char *p;
 		return true;
 	return false;
 }
+
+/**
+ * Create scema of tables
+ *
+ * @param payload   The  payload containing information about schema of 
+ *                  tables to create
+ * @return true if the tables can be crated successfully
+ */
+int Connection::create_schema(std::string payload)
+{
+
+	Document document;
+	std::string schema;
+	int version;
+	const char *logSection="CreatingSchema";
+	unsigned long rowsAffectedLastCommand = 0;
+
+	std::string queryToFindVerFromDB = "select * from fledge.version;" ;
+	int versionFromDB = purgeOperation(queryToFindVerFromDB.c_str(), logSection, "logSection- finding version stored in db", true);
+	Logger::getLogger()->debug("%s:%d Version obtained from DB %d ", __FUNCTION__,__LINE__ , versionFromDB);
+
+	bool schemaUpdateRequired = false;	
+
+	try {
+                if (payload.empty())
+                {
+                        return -1;
+                }
+                else
+                {
+                        if (document.Parse(payload.c_str()).HasParseError())
+                        {
+                                raiseError("retrieve", "Failed to parse JSON payload");
+                                return -1;
+                        }
+			if (!document.HasMember("schema"))
+                        {
+				Logger::getLogger()->error("schema absent from payload, so exiting ");
+				return -1;
+			}
+			else
+			{
+				schema = document["schema"].GetString();
+				if ( !schema.empty())
+				{
+					std::string queryToCreateSchema = "create schema if not exists " + schema + ";" ;
+					rowsAffectedLastCommand = purgeOperation(queryToCreateSchema.c_str(), logSection, "Create Schema if not exists ", false);
+				}
+			/*	if ( !schema.empty())
+				{
+					std::string queryToFindSchema = "select count(*) from fledge.service_schema where name = " + schema +";" ;
+					rowsAffectedLastCommand = purgeOperation(queryToFindSchema.c_str(), logSection, "Create Schema finding schema name from fledge.service_schema", true);
+
+					if (rowsAffectedLastCommand != 1)
+					{
+						schemaUpdateRequired = true;
+					}
+				}
+			*/
+				if (!document.HasMember("version"))
+                        	{
+					Logger::getLogger()->debug("version absent from payload, so exiting");
+                                	return -1;
+                        	}
+				else
+				{
+					version = document["version"].GetInt();
+					if ( version == versionFromDB)
```

Comment:
As above this is not the right check as the fledge schema version ID is not the same as the version of the schema you are working with in this API call.

Reply:
resolved